File: py/pages/login_page.py
from selenium.webdriver.common.by import By
from .base_page import BasePage
from config import BASE_URL

class LoginPage(BasePage):
    USERNAME = (By.ID, "username")
    PASSWORD = (By.ID, "password")
    LOGIN_BTN = (By.ID, "kc-login")

    def load(self):
        self.open(BASE_URL)

    def login(self, username, password):
        self.find(*self.USERNAME).send_keys(username)
        self.find(*self.PASSWORD).send_keys(password)
        self.find(*self.LOGIN_BTN).click()

File: kt/src/main/kotlin/pages/LoginPage.kt
package pages

import org.openqa.selenium.By
import org.openqa.selenium.WebDriver
import org.openqa.selenium.support.ui.ExpectedConditions
import org.openqa.selenium.support.ui.ExpectedConditions.presenceOfElementLocated
import org.openqa.selenium.support.ui.WebDriverWait
import java.time.Duration

class LoginPage(private val driver: WebDriver) {
    private val wait = WebDriverWait(driver, Duration.ofSeconds(10))

    fun load(url: String) {
        driver.get(url)
        wait.until(ExpectedConditions.presenceOfElementLocated(By.id("username")))
    }

    fun login(username: String, password: String) {
        driver.findElement(By.id("username")).sendKeys(username)
        driver.findElement(By.id("password")).sendKeys(password)
        driver.findElement(By.id("kc-login")).click()
    }
}
